lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,187 @@
+  end
+
+  def self.plugins_with_locales
+    configuration[:plugins]
+  end
+
+  def self.configuration
+    # todo raise error if something is missing
+    @configuration ||= begin
+      configuration_file = Pathname(__FILE__) + '../../../../../'
+    configuration_file = configuration_file + 'configuration.yml'
+    YAML.load_file(configuration_file)
+    end
+  end
+
+  def self.set_crowdin_specifics(configuration_hash)
+    @project_id = configuration_hash[:project_id]
+    @api_key = configuration_hash[:api_key]
+    @crowdin_directory = configuration[:crowdin_directory]
+  end
+
+  def self.branch
+    configuration[:branch]
+  end
+
+  def self.previous_branch
+    configuration[:previous_branch]
+  end
+
+  def self.upload_english
+    # create crowdin directory just in case it doesn't exist.
+    crowdin = create_crowdin_handle
+    dir = crowdin.project_info['files'].find {|f| f['name'] == @crowdin_directory && f['node_type'] == 'directory'}
+    unless dir
+      crowdin.add_directory(@crowdin_directory)
+    end
+
+    # either add or update the english translation file
+    path_to_translation = File.join 'config', 'locales', 'en.yml'
+    dir = crowdin.project_info['files'].find {|f| f['name'] == @crowdin_directory && f['node_type'] == 'directory'}
+    if dir['files'].find {|f| f['name'] == 'en.yml'}
+      crowdin.update_file [{dest: "/#{@crowdin_directory}/en.yml", source: path_to_translation.to_s, title: 'OpenProject wording', export_pattern: '%two_letters_code%.yml'}], type: 'yaml'
+    else
+      crowdin.add_file [{dest: "/#{@crowdin_directory}/en.yml", source: path_to_translation.to_s, title: 'OpenProject wording', export_pattern: '%two_letters_code%.yml'}], type: 'yaml'
+    end
+
+    # either add or update the english javascript translation file
+    path_to_translation = File.join 'config', 'locales', 'js-en.yml'
+    dir = crowdin.project_info['files'].find {|f| f['name'] == @crowdin_directory && f['node_type'] == 'directory'}
+    if dir['files'].find {|f| f['name'] == 'js-en.yml'}
+      crowdin.update_file [{dest: "/#{@crowdin_directory}/js-en.yml", source: path_to_translation.to_s, title: 'OpenProject JavaScript wording', export_pattern: 'js-%two_letters_code%.yml'}], type: 'yaml'
+    else
+      crowdin.add_file [{dest: "/#{@crowdin_directory}/js-en.yml", source: path_to_translation.to_s, title: 'OpenProject JavaScript wording', export_pattern: 'js-%two_letters_code%.yml'}], type: 'yaml'
+    end
+  end
+
+  def self.request_build
+    # todo maybe this could run into a timeout?
+    crowdin = create_crowdin_handle
+    crowdin.export_translations
+  end
+
+  def self.create_crowdin_handle
+    Crowdin::API.new project_id: @project_id, api_key: @api_key
+  end
+
+  def self.download_and_replace_locales
+    # todo delete all locales here? maybe for the case that
+    # we do not support a language anymore.
+    begin
+      languages_files = create_temp_file('crowdin_translations')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.